### PR TITLE
Add TEP-0024: Embedded Trigger Templates

### PR DESCRIPTION
**Pull request by dibyom**
Triggers issue: https://github.com/tektoncd/triggers/issues/616

### CODE REVIEW

**Review by wlynch — COMMENTED** [review]
/lgtm

No objections here!
Only nit is it would be great to see some inline examples!

**Comment by dibyom** [comment]
Added an example!

File: teps/0024-embedded-trigger-templates.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        spec:
+          taskSpec:
+            steps:
+            - image: ubuntu
+              script: echo "hello there $(tt.params.name)"
+  ```
+
+2. Deprecate the `name` field for reffering to TriggerTemplate objects in favor
+   of a new `ref` field. This is for consistency as we use usually use `ref` to
+   refer to other resources (see `bindings` as an example). Example:
```
Comment (savitaashture) [inline teps/0024-embedded-trigger-templates.md:78]: Can we provide link to bindings?

**Review by savitaashture — COMMENTED** [review]
/lgtm 

small nit to address if it seems valid to you

**Comment by sbwsg** [comment]
Nice.

/lgtm

File: teps/0024-embedded-trigger-templates.md (hunk 1 above)
Comment (dibyom) [reply teps/0024-embedded-trigger-templates.md:78]: ahh nice catch :)

**Review by savitaashture — COMMENTED** [review]
/lgtm

**Review by wlynch — APPROVED** [review]
/lgtm

**Pull request merged**